services/placement/algo/helper.go

```diff
@@ -490,8 +490,8 @@ func addInstanceToPlacement(p services.ServicePlacement, i services.PlacementIns
 }
 
 func removeInstanceFromPlacement(p services.ServicePlacement, id string) (services.ServicePlacement, services.PlacementInstance, error) {
-	leavingInstance := p.Instance(id)
-	if leavingInstance == nil {
+	leavingInstance, exist := p.Instance(id)
+	if !exist {
 		return nil, nil, errInstanceAbsent
```

Comment:
hm might be better to return the same error in `MarkShardAvailable` if the instance is not found?